--- src/gcc-6.2.0/gcc/ubsan.c	2016-08-12 15:57:35.000000000 -0400
+++ src/gcc-6.2.0/gcc/ubsan.c	2021-08-31 20:44:55.000000000 -0400
@@ -1469,7 +1469,7 @@
 
   expanded_location xloc = expand_location (loc);
   if (xloc.file == NULL || strncmp (xloc.file, "\1", 2) == 0
-      || xloc.file == '\0' || xloc.file[0] == '\xff'
+      || xloc.file[0] == '\0' || xloc.file[0] == '\xff'
       || xloc.file[1] == '\xff')
     return false;
 
